### PR TITLE
Merge Parachain

### PR DESCRIPTION
**Pull request by weichweich**
## fixes KILTProtocol/ticket#896

- we have a separate Node and Runtime project for the Mashnet and the Parachain
- Folder structure
  - on the first level we have node and runtime and on the second level the specific Parachain and Mashnet crates
  - could be the other way around
  - reason for this way: fist level is project type (pallet, node, runtime, primitives), second level is more specific
- Dockerfile
  - @rflechtner SORRY! removed the dependency caching...
  - produced an issue while building
  - IMHO complex Dockerfile&maintainance outweigh the benefits
  - huh building takes ages now! 😨

### CODE REVIEW

File: nodes/parachain/src/rpc.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+/// Instantiate all full RPC extensions.
+pub fn create_full<C, P>(deps: FullDeps<C, P>) -> jsonrpc_core::IoHandler<sc_rpc::Metadata>
+where
+	C: ProvideRuntimeApi<Block>,
+	C: HeaderBackend<Block> + HeaderMetadata<Block, Error = BlockChainError> + 'static,
+	C: Send + Sync + 'static,
+	C::Api: frame_rpc_system::AccountNonceApi<Block, AccountId, Index>,
+	// C::Api: pallet_transaction_payment_rpc::TransactionPaymentRuntimeApi<Block, Balance>,
```
Comment (wischli) [inline nodes/parachain/src/rpc.rs:34]: Is this left out intentionally?

File: nodes/parachain/src/rpc.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	C::Api: BlockBuilder<Block>,
+	P: TransactionPool + 'static,
+{
+	use frame_rpc_system::{FullSystem, SystemApi};
+	// use pallet_transaction_payment_rpc::{TransactionPayment, TransactionPaymentApi};
```
Comment (wischli) [inline nodes/parachain/src/rpc.rs:39]: Is this left out intentionally?

File: nodes/parachain/src/rpc.rs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		deny_unsafe,
+	} = deps;
+
+	io.extend_with(SystemApi::to_delegate(FullSystem::new(
+		client.clone(),
+		pool,
+		deny_unsafe,
+	)));
+
+	// io.extend_with(TransactionPaymentApi::to_delegate(TransactionPayment::new(
```
Comment (wischli) [inline nodes/parachain/src/rpc.rs:54]: Is this left out intentionally?

**Review by wischli — APPROVED** [review]
LGTM if the code snippets (see below) were commented out intentionally

File: nodes/parachain/src/rpc.rs (hunk 1 above)
Comment (weichweich) [reply nodes/parachain/src/rpc.rs:34]: tracked in KILTprotocol/ticket/issues/1039

File: nodes/parachain/src/rpc.rs (hunk 2 above)
Comment (weichweich) [reply nodes/parachain/src/rpc.rs:39]: tracked in KILTprotocol/ticket/issues/1039

File: nodes/parachain/src/rpc.rs (hunk 3 above)
Comment (weichweich) [reply nodes/parachain/src/rpc.rs:54]: tracked in KILTprotocol/ticket/issues/1039

**Pull request merged**